packages/frontend/src/hooks/useDefiLlamaCurrentTvl.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+
+export const useDefiLlamaCurrentTvl = (
+    protocol?: string,
+): {
+    loading: boolean;
+    tvl: string;
+} => {
+    const [loading, setLoading] = useState(false);
+    const [tvl, setTvl] = useState("");
+
+    useEffect(() => {
+        let cancelled = false;
+        const fetchData = async (): Promise<void> => {
+            if (!protocol) return;
+            if (!cancelled) setLoading(true);
+            try {
+                const response = await fetch(
+                    `https://api.llama.fi/tvl/${protocol}`,
+                );
+                if (!response.ok) {
+                    console.error(
+                        `could not fetch current tvl for protocol ${protocol} from defillama`,
+                    );
+                    return;
```

Comment:
Should we set `loading` to `false` here? Otherwise, in case of a response error, it would load forever.

Reply:
The `finally` block should already do that